xdsl/ir.py

```diff
@@ -710,9 +706,8 @@ def insert_arg(self, typ: Attribute, index: int) -> BlockArgument:
         new_arg = BlockArgument(typ, self, index)
         for arg in self._args[index:]:
             arg.index += 1
-        self._args = FrozenList(
-            list(self._args[:index]) + [new_arg] + list(self._args[index:]))
-        self._args.freeze()
+        self._args = tuple(
+            chain(self._args[:index], [new_arg], self._args[index:]))
```

Comment:
I am not very versed with tuples, so maybe a dumb question:
Why did you change the way this was done? You can create a tuple from a list, right? Why the switch to chain?

Reply:
you can't `+` tuples, but you can construct them from an iterable. The `chain` is more idiomatic as it involves way fewer copies of elements to temporary lists

Reply:
But you can create a tuple from lists that you `+`ed right?
So `tuple([1, 2, 3] + [4, 5, 6])` should actually work?

Reply:
yep